stage1/init/common/pod.go

```diff
@@ -565,6 +565,50 @@ func PodToSystemd(p *stage1commontypes.Pod, interactive bool, flavor string, pri
 	return nil
 }
 
+// evaluateAppMountPath tries to resolve symlinks within the path.
+// It returns the actual relative path for the given path.
+// TODO(yifan): This is a temporal fix for systemd-nspawn not handling symlink mounts well.
+func evaluateAppMountPath(appRootfs, path string) (string, error) {
+	link := appRootfs
+
+	paths := strings.Split(path, "/")
+	for i, p := range paths {
```

Comment:
Should we skip empty strings?

```
if p == "" {
  continue
} 
```

It can happen because `strings.Split("/a/b", "/")` returns `["" "a" "b"]`.

I think empty strings would mess up the `os.Readlink(next)` below.